crates/runtime/src/virtual_machine.rs

```diff
@@ -158,85 +117,104 @@ impl VirtualMachine {
                     })
             })
             .collect();
-        let prepare_for_lines_handler = self.prepare_for_lines_handler.as_mut().unwrap();
-        prepare_for_lines_handler.call(string_ids, &mut self.handler_safe_dialogue);
+        self.events.push(DialogueEvent::PrepareForLines(string_ids));
     }
 
     /// Resumes execution.
-    pub(crate) fn continue_(&mut self) {
-        self.assert_can_continue();
-        if *self.execution_state() == ExecutionState::DeliveringContent {
-            // We were delivering a line, option set, or command, and
-            // the client has called Continue() on us. We're still
-            // inside the stack frame of the client callback, so to
-            // avoid recursion, we'll note that our state has changed
-            // back to Running; when we've left the callback, we'll
-            // continue executing instructions.
-            *self.execution_state_mut() = ExecutionState::Running;
-            return;
+    #[must_use]
+    pub(crate) fn continue_(&mut self) -> Option<Vec<DialogueEvent>> {
```

Comment:
Perhaps more idiomatic (if this works for function names):
```suggestion
    pub(crate) fn r#continue(&mut self) -> Option<Vec<DialogueEvent>> {
```

Reply:
[`e885283` (#103)](https://github.com/yarn-slinger/yarn-slinger/pull/103/commits/e885283a9c369a29a3cf3c315a5689c214a54059)